goshopify_test.go

```diff
@@ -44,7 +44,7 @@ func setup() {
 		Scope:       "read_products",
 		Password:    "privateapppassword",
 	}
-	client = NewClient(app, "fooshop", "abcd", WithVersion(testApiVersion))
+	client = NewClient(app, "fooshop", "abcd", WithVersion(testApiVersion), WithRetry(3))
```

Comment:
This is not really testing the retry functionality, I think you need a bit more elaborate test that has a http mock that returns one or two times a 429 and then succeeds (and check that you got all the requests that you expected).

Reply:
the retry logic is tripped here: ttps://github.com/bold-commerce/go-shopify/blob/fa02f8527c483bbc6ab25aa0b8d6428611001bb0/goshopify_test.go#L330

during testing I saw 3 backoff and retries

Reply:
Right but I don't see any code checking for the expected behavior of the API client.
Also, you'd want to check a few more scenarios, e.g. `withRetry(3) and returning a 429 twice` will result in 3 requests and eventual success, `withRetry(3) and returning a 429 three times` will result in an error, etc

Reply:
sure no problem ill update that.